### PR TITLE
Add the Sqitch app for database management

### PR DESCRIPTION
**Pull request by robinbb**

### CODE REVIEW

File: pkgs/development/tools/misc/sqitch/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, writeScript, makeWrapper, sqitchModule }:
+stdenv.mkDerivation rec {
+  name = "sqitch-0.999";
+  buildInputs = [ makeWrapper ];
+  propagatedNativeBuildInputs = [ sqitchModule ];
+  builder = writeScript (name + "-builder.sh") ''
```
Comment (lethalman) [inline pkgs/development/tools/misc/sqitch/default.nix]: Can you avoid using the builder here? I think this is enough:
```
unpackPhase = "true";
buildPhase = "true";
installPhase = those symlinks;
postFixup = the wrapper;
```

Also in the wrapper you shouldn't need to do that for by yourself, just `--prefix PERL5LIB $PERL5LIB`. And sqitchModule doesn't need to be propagated I think.

Comment (robinbb) [inline pkgs/development/tools/misc/sqitch/default.nix]: I have removed the unnecessary propagatedNativeBuildInputs line in my latest commit.

Regarding PERL5LIB: it is not set within the stdenv.mkDerivation builder, so my invocation of wrapProgram has no access to it. It must be deduced based on the ${sqitchModule}/nix-support/propagated-native-build-inputs file. If there's a better way, please let me know.

**Comment by robinbb** [comment]
@lethalman I have incorporated your comments in commit a3a98d7. Thank you for your helpful input. Please review.

**Comment by lethalman** [comment]
I'm totally confused. There's already an AppSqitch in perl-packages.nix in git master, why this PR adds another one?

**Comment by robinbb** [comment]
@lethalman perhaps you are looking at the diff between this branch and 'master' at the time that this PR was opened, instead of this branch and current 'master'. This branch does not add an AppSqitch module. That was done in an already-merged PR. Try examining with 'git diff -w master..a3a98d7'.

**Comment by lethalman** [comment]
I'm looking at the "Files changed" tab of this PR. Perhaps cleanup this PR by rebasing your changes against the new master.

**Pull request closed without merging**